src/coreclr/jit/importer.cpp

```diff
@@ -15115,9 +15115,20 @@ void Compiler::impImportBlockCode(BasicBlock* block)
                     {
                         assert(aflags & CORINFO_ACCESS_GET);
 
-                        LPVOID         pValue;
-                        InfoAccessType iat = info.compCompHnd->emptyStringLiteral(&pValue);
-                        op1                = gtNewStringLiteralNode(iat, pValue);
+                        // Try to import String.Empty as "" (GT_CNS_STR) if VM has a mdToken for ""
+                        uint32_t emptyStrTok = info.compCompHnd->getEmptyStringMdToken();
```

Comment:
This is not pretty. Why is the metadata token needed to make the JIT happy? Can the JIT assume that e.g. token with value of `0` means empty string instead?

Reply:
@jkotas what you suggest was my initial implementation: https://github.com/dotnet/runtime/pull/44847/commits/4f9d795f02606ddbb2f2d061c7e6585735f8aa4d

in this PR I'm addressing https://github.com/dotnet/runtime/pull/44847#pullrequestreview-556647573 cc @AndyAyersMS

Reply:
> what you suggest was my initial implementation: 4f9d795

I think that was better.

In general, methods on JIT/EE interface that return tokens are anti-pattern. The JIT/EE interface should always go from token to runtime entity like handle, pointer or offset.

Reply:
The main problem in the original implementation was the static cache. I think you can replace the static cache by picking up a number that cannot be a valid string token, like `0` or `-1`.

Reply:
Switched to JIT-only change (SconCPX == 0), previous impl is here https://github.com/EgorBo/runtime-1/commit/15a47c2660fe5b9e2149b20361a6f6645575d66e